src/items/rqgItem.ts

```diff
@@ -96,6 +100,101 @@ export class RqgItem extends Item {
     });
   }
 
+  protected _onCreate(
+    data: ItemDataSource,
+    options: DocumentModificationOptions,
+    userId: string
+  ): void;
+  protected _onCreate(
+    data: RqgItem["data"]["_source"],
+    options: DocumentModificationOptions,
+    userId: string
+  ): void {
+
+    if (data.type === ItemTypeEnum.Armor) {
+      //@ts-ignore
+      const item = getGame().items?.get(data._id);
+      item?.update({
+        img: String(getGame().settings.get("rqg", "defaultIconArmor")),
+        "data.namePrefix": data.name,
+      });
+    }
```

Comment:
If we stored _one_ defaultIcons setting with an object like
```ts
{
[ItemTypeEnum.Armor]: "systems/rqg/assets/images/armor/cuirass.svg",
[ItemTypeEnum.Cult]: "systems/rqg/assets/images/items/cult.svg",
...
}
```
Then this repetition could be simplified to something like

```ts
const defaultIcons = getGame().settings.get("rqg", "defaultIcons"));
//@ts-ignore
const item = getGame().items?.get(data._id);
item?.update({
  img: defaultIcons[data.type],
});
```
And the settings for icons would be grouped into a separate dialog. They are quite many and overpowers the system settings  a bit. This would require creating a default icons settings Dialog though. I haven't looked in to how complicated that is...

The (very good) fix for item names would still require an "if" I guess for the item types that need it.

Reply:
I agree that it junks up the settings.  I couldn't find some way to launch a custom dialog from the settings.  Or would we have to have some other way to launch that dialog?  Or are you suggesting just one text setting with a JSON blob in it?

Reply:
I haven't looked into it much myself, but this describes how to set it up. https://foundryvtt.wiki/en/development/api/settings#setting-menus
I think the actual setting can be an Object and the setting FormApplication would just show a list of filepickers - one for each itemtype.